javabuilder-authorizer/token_validator.rb

```diff
@@ -25,11 +26,15 @@ def validate
     return error(TOKEN_USED) unless log_token
     return error(USER_BLOCKED) if user_blocked?
     return error(TEACHERS_BLOCKED) if teachers_blocked?
-    return error(USER_OVER_HOURLY_LIMIT) if user_over_hourly_limit?
+    hourly_usage_response = user_usage(ONE_HOUR_SECONDS)
+    return error(USER_OVER_HOURLY_LIMIT) if user_over_hourly_limit?(hourly_usage_response)
     return error(USER_OVER_DAILY_LIMIT) if user_over_daily_limit?
     return error(TEACHERS_OVER_HOURLY_LIMIT) if teachers_over_hourly_limit?
+    near_limit_detail = user_near_hourly_limit?(hourly_usage_response.count)
+
     log_requests
     mark_token_as_vetted
+    set_token_warning(NEAR_LIMIT, near_limit_detail) if near_limit_detail
```

Comment:
I know we're not planning on warning on daily limit yet, but just curious, will we only show one warning if the user is near both their hourly and daily limit?

Reply:
yeah right now we are set up to only have one warning (we would need multiple warning fields if we wanted to show multiple warnings). In theory it's pretty unlikely someone would reach both their hourly and daily at the same time, and the messaging would be similar (wait to run your code again).